lib/internal/bootstrap/node.js

```diff
@@ -150,6 +150,16 @@ if (isMainThread) {
   setupProcessStdio(getStdout, getStdin, getStderr);
 }
 
+// Setup the callbacks that node::AsyncWrap will call when there are hooks to
+// process. They use the same functions as the JS embedder API. These callbacks
+// are setup immediately to prevent async_wrap.setupHooks() from being hijacked
+// and the cost of doing so is negligible.
+const { nativeHooks } = require('internal/async_hooks');
+internalBinding('async_wrap').setupHooks(nativeHooks);
+
+// XXX(joyeecheung): this has to be done after the initial load of
+// `internal/async_hooks` otherwise `async_hooks` cannot require
+// `internal/async_hooks`. Investigate why.
```

Comment:
These three lines of comments can be removed now?

Reply:
I don't think the lines added above answers this question..the bug seems to lie in the module loading order or the scope resolution somehow. So I'd prefer to keep it until we figure out why..

Reply:
Looks like the order of calling `setupHooks` is not the key here: https://ci.nodejs.org/job/node-stress-single-test/2192/ for https://github.com/nodejs/node/commit/cede98ce522089e2937f843d7508906b38c0571a is still green. (though it improves the spaghetti a bit..)